From eacc41c1ca0c36218698058b3f8954549952efee Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sun, 17 Mar 2024 22:44:57 -0700
Subject: [PATCH 08/10] i40e: Include xdp.h where required

This commit accommodates commit 680ee0456a57 ("net: invert the netdevice.h
vs xdp.h dependency") which removes the "#include <xdp.h>" C preprocessor
directive from header file "netdevice.h". This commit was merged in the
v6.6 development cycle and causes compilation failures with the v6.6 kernel,
which are listed below.

We accommodate this commit by including the "xdp.h" header file as needed
from "i40e_txrx.h".

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=680ee0456a57

Resolved compilation errors:

src/i40e_txrx.h:467:22: error: field xdp_rxq has incomplete type
  467 |  struct xdp_rxq_info xdp_rxq;

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/i40e_txrx.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/i40e_txrx.h b/src/i40e_txrx.h
index f14c1ba..344de09 100644
--- a/src/i40e_txrx.h
+++ b/src/i40e_txrx.h
@@ -4,6 +4,8 @@
 #ifndef _I40E_TXRX_H_
 #define _I40E_TXRX_H_
 
+#include <net/xdp.h>
+
 /* Interrupt Throttling and Rate Limiting Goodies */
 #define I40E_DEFAULT_IRQ_WORK      256
 
-- 
2.42.0

